tensorboard/components/tf_data_selector/tf-data-selector.ts

```diff
@@ -69,6 +82,43 @@ Polymer({
         newComparingExpIds);
   },
 
+  _canCompareExperiments(): boolean {
+    return Boolean(this._comparingExps.length);
+  },
+
+  /**
+   * Prunes away an experiment that has been removed from `_comparingExps` from
+   * the _selectionMap.
+   */
+  _pruneSelection() {
+    if (!this._canCompareExperiments()) return;
+
+    const expLookupMap = new Map(this._comparingExps.map(e => [e.id, e]));
+    Array.from(this._selectionMap.keys()).forEach(expId => {
+      if (!expLookupMap.has(expId)) this._selectionMap.delete(expId);
+    });
+
+    this.set('selection', Array.from(this._selectionMap.values()));
+  },
+
+  _selectionChanged(event) {
+    const {runs, tagRegex} = event.detail;
+    const expLookupMap = new Map(this._comparingExps.map(e => [e.id, e]));
+    const expId = this._canCompareExperiments() ?
```

Comment:
Optional, but in the no-experiments case, rather than defining expId to be null and then not using the value, it seems simpler to just handle it separately like

```js
if (!this._canCompareExperiments()) {
  this.set('selection', [{runs, tagRegex}]);
  return;
}
const expId = event.target.experiment.id;
this._selectionMap.set(expId, ...);
this.set('selection', Array.from(this._selectionMap.values()));
```